sql/parse/parse.go

```diff
@@ -1711,18 +1720,33 @@ func convertCreateTable(ctx *sql.Context, c *sqlparser.DDL) (sql.Node, error) {
 			return nil, err
 		}
 
+		// TODO: helper method
+		var hasBlob bool
+		prefixLengths := make([]uint16, len(columns))
+		for i, col := range columns {
+			if blobCols[col.Name] {
+				hasBlob = true
+			}
+			prefixLengths[i] = uint16(col.Length)
+		}
+		if !hasBlob {
+			prefixLengths = nil
+		}
```

Comment:
Rather than allocate an array that is discarded for the majority of types, it makes more sense to have it as `nil` and allocate the array when necessary. So this can just be `var prefixLengths []uint16`. In the loop, just check if the array is nil and create one before assigning to an index. Something like
```go
var prefixLengths []uint16
for i, col := range columns {
    if blobCols[col.Name] {
        if len(prefixLengths) == 0 {
            prefixLengths := make([]uint16, len(columns))
        }
        prefixLengths[i] = uint16(col.Length)
    }
}
```